### DISCUSSION TITLE
Disable ansi codes for logs

### DISCUSSION THREAD

**Discussion by zlobniyshurik** [meilisearch/product#764] [Feedback & Feature Proposal]
My system use classic text logs. So ansi codes in logs are problem for me.
Now my logs look like this:
```
.[2m2024-07-10T01:56:51.049606Z.[0m .[32m INFO.[0m .[2mindex_scheduler.[0m.[2m:.[0m A batch of tasks was successfully completed with 1 successful tasks and 0 failed tasks.
```
Can I choose clear text log format without any fancy things?

**Comment by WKoster** [comment]
Mine look like #011 and #033[2m etc... 

So in the mean time I filter them out with something like:

sed -i 's/#011//g'          $log
sed -i 's/#033\[2m//g'  $log
sed -i 's/#033\[0m//g'  $log
sed -i 's/#033\[32m//g' $log
sed -i 's/#033\[34m//g' $log

or probably even with 1 line of code.

note: this (-i) changes the logfile itself!

**Comment by zlobniyshurik** [comment]
At now I use ugly hack:
Systemd-service, that start **meilisearch**: 
***/etc/systemd/system/meilisearch.service***
```
[Unit]
Description=Meilisearch Service
Requires=opt-meilisearch.mount
After=opt-meilisearch.mount

[Service]
User=meilisearch
Group=meilisearch
WorkingDirectory=/opt/meilisearch
#StandardOutput=journal
StandardOutput=append:/var/log/meilisearch/meilisearch.log
StandardError=append:/var/log/meilisearch/meilisearch.log
ExecStart=/bin/bash -c /opt/meilisearch/start_meili.sh

[Install]
WantedBy=multi-user.target
```
and bash-script for output filtering:
***/opt/meilisearch/start_meili.sh***
```bash
#!/usr/bin/env bash

cd /opt/meilisearch
exec 1> >(stdbuf -oL sed -u -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2};?)?)?[mGK]//g")
exec 2>&1
meilisearch --config-file-path="./config.toml"
```
Far from perfect but its working :)

P.S. In an ideal world, **meilisearch** would have a setting in its config file to disable color output.

**Reply by curquiza** [reply]
Hello everyone, I forwarded your feedback to @meilisearch/product-team

**Reply by macraig** [reply]
Hello! I've transferred the feature request for disabling ansi codes for logs to the Product Feedback & Feature request discussions section to make it visible for other users and gather more requests.

**Comment by szepeviktor** [comment]
You can use the `ansi2txt` command from the [colorized-logs](https://packages.debian.org/bookworm/colorized-logs) package (as a work-around).

**Comment by dotkas** [comment]
Could this work? https://www.meilisearch.com/docs/learn/self_hosted/configure_meilisearch_at_launch#customize-log-output
